### ISSUE TITLE
Unhandled exception when checking lambdas

### ISSUE THREAD

**Issue by chpeer** [google/error-prone#628]
Hi, the following code snippet produces an error
```
producer.send(data, (recordMetadata, e) -> log.error("test"));
```
error:
```
An unhandled exception was thrown by the Error Prone static analysis plugin.
     Please report this at https://github.com/google/error-prone/issues/new and include the following:

       error-prone version: 2.0.14
     Stack Trace:
     com.sun.tools.javac.code.Symbol$CompletionFailure: class file for org.apache.kafka.clients.producer.RecordMetadata not found
```

If I remove the lambda expression, error-prone does not complain
```
producer.send(data, new Callback() {
      @Override
      public void onCompletion(RecordMetadata recordMetadata, Exception e) {
        log.error("test");
      }
    });
```

Producer is of type `org.apache.kafka.clients.producer.KafkaProducer`

**Comment by abhinavrpatel** [comment]
Is this still an issue? If so, I would love to look into it.

**Comment by eaftan** [comment]
@abhinavrpatel We're not sure if it's still an issue, but we would love help. :)  I would start by trying to reproduce the bug on version 2.0.14 of Error Prone, then see if it still repros with the latest version.  

Thanks!
